pkg/build/templates/coredns/deployment-coredns.yaml

```diff
@@ -0,0 +1,126 @@
+apiVersion: apps/v1
+kind: Deployment
```

Comment:
As kind installs OOTB CoreDNS, why do also install it using idpbuilder ? AFAIK It is only needed to patch the coredns ConfigMap !

Reply:
I don' think this was clear, we are not installing them. We are just updating the spec around mounting config maps. 

A few points about CoreDNS I should make clear:
1. CoreDNS file allows you to import files matching a pattern.
2. By default, CoreDNS installation only mounts the corefile key within coredns CM.

Taken together, what we need to do to allow our default customization and possible further customization:
1. Update the coredns CM to use the import plugin, then make it look for files matching `/etc/coredns-configs/*.conf`
2. Populate `/etc/coredns-configs/` with two files (sourced from CMs):
    - One that contains our default configuration. That is to resolve the name given by the host flag and all subdomains to the internal ingress address.
    - One that is empty. This is to allow for other packages to update this file for further DNS configurations. In the absence of this file, other packages must update the coredns deployment.
3. Update the coredns deployment to mount the two CMs.

Reply:
This needs to be documented. It is a common practice I think? AKS does something very similar and other k8s provider do it too. https://learn.microsoft.com/en-us/azure/aks/coredns-custom